build/README.md

```diff
@@ -14,188 +16,380 @@ To run a local DSS instance for testing, evaluation, or development, see [dev/st
+        are the recommended instructions (`gcloud auth configure-docker`).
+        Ensure that
+        [appropriate permissions are enabled](https://cloud.google.com/container-registry/docs/access-control).
+
+1. Use the [`build.sh` script](./build.sh) in this directory to build and push
+   an image tagged with the current date and git commit hash.
+        
+1. Note the image name (e.g., `gcr.io/your-project-id/dss:2020-07-10-9cc0a6a84`)
+   printed at the end of the script.
+
 ## Deploying the DSS on Kubernetes
 
 Note: All DSS instances in the same cluster must point their ntpd at the
 same NTP Servers. [CockroachDB recommends](https://www.cockroachlabs.com/docs/stable/recommended-production-settings.html#considerations)
 using [Google's Public NTP](https://developers.google.com/time/) when
 running in a multi-cloud environment.
 
-This secion discusses deploying a Kubernetes service, although you can deploy
+This section discusses deploying a Kubernetes service, although you can deploy
 a DSS instance however you like as long as it meets the CockroachDB requirements
 above. You can do this on any supported
 [cloud provider](https://kubernetes.io/docs/concepts/cluster-administration/cloud-providers/)
 or even on your own infrastructure. Consult the Kubernetes documentation for
-which ever provider you choose.
-
-1.  Create a new Kubernetes cluster as mentioned above. We recommend a new
-    cluster for each DSS instance.
-
-1.  Create a namespace for your Kubernetes cluster in which the DSS services will reside. This can be achieved with `kubectl create ns <NAMESPACE>`. Alternatively, you may want deploy in a namespace that already exists, so keep track of the namespace for later steps. The namespace name does not matter, though it will affect the DNS addresses associated with the certificates generated. One convention might be to reference a namespace named `production` to differentiate another cluster namespace called `development`. Another might be to name it `dss` to differentiate these services from the other namespaces within the cluster.
-
-1.  Create your static IP addresses. How you do this depends on your cloud
-    provider:
-
-    - 1 for the HTTPS Gateway's Ingress. Note: if you're using Google Cloud
-      the HTTPS Gateway Ingress needs to be created as a "Global" IP address.
-    - If you want to be able to join other clusters, you'll need static IP's
-      for each of your CockroachDB nodes (min of 3), and you'll need to create
-      DNS entries with a common hostname suffix, ie: `*.db.interuss.com`
-
-1.  Use the `make-certs.py` script in this directory to create certificates for
+your chosen provider.
+
+If you can augment this documentation with specifics for another cloud provider,
+a PR to that effect would be greatly appreciated.
+
+1.  Create a new Kubernetes cluster. We recommend a new cluster for each DSS
+    instance.  A reasonable cluster name might be `dss-us-prod`,
```

Comment:
nit: why are we providing naming suggestions? If we are we might want to put more thought into the suggestions themselves. I would likely recommend including the availability zone in the name as well

Reply:
To illustrate good practice and imply a set of rules a reasonable name should follow without actually listing out those rules -- we want someone to be able to follow these instructions with minimal interpretation or additional knowledge necessary and still end up with a good DSS instance.

Including zone sounds good; would you recommend `dss-us-us-east4-a-prod` or `dss-us-east4-a-prod` or `dss-us-prod-us-east-4a` or something else?

Reply:
anything unique to the AZ sounds good to me. I'm partial to short names like `us-e4a`, resulting in `dss-prod-us-e4a`

Reply:
Sounds good; done, thanks.